File: android/src/main/kotlin/com/tradechart/plugin/bridge/ChartFlutterApiHolder.kt
package com.tradechart.plugin.bridge

import com.tradechart.plugin.bridge.generated.ChartFlutterApi
import com.tradechart.plugin.bridge.generated.CrosshairDataMessage
import com.tradechart.plugin.bridge.generated.ViewportStateMessage

class ChartFlutterApiHolder(
    private val flutterApi: ChartFlutterApi,
) {
    fun onChartReady() {
        flutterApi.onChartReady {}
    }

    fun onViewportChanged(viewport: ViewportStateMessage) {
        flutterApi.onViewportChanged(viewport) {}
    }

    fun onCrosshairData(data: CrosshairDataMessage) {
        flutterApi.onCrosshairData(data) {}
    }

    fun onError(code: String, message: String) {
        flutterApi.onError(code, message) {}
    }
}

File: ios/Classes/Bridge/ChartFlutterApiHolder.swift
import Foundation

final class ChartFlutterApiHolder {
  init(flutterApi: ChartFlutterApi) {
    self.flutterApi = flutterApi
  }

  private let flutterApi: ChartFlutterApi

  func onChartReady() {
    flutterApi.onChartReady { _ in }
  }

  func onViewportChanged(_ viewport: ViewportStateMessage) {
    flutterApi.onViewportChanged(viewport: viewport) { _ in }
  }

  func onCrosshairData(_ data: CrosshairDataMessage) {
    flutterApi.onCrosshairData(data: data) { _ in }
  }

  func onError(code: String, message: String) {
    flutterApi.onError(code: code, message: message) { _ in }
  }
}
